wp-includes/class-wp-service-workers.php

```diff
@@ -239,206 +264,83 @@ protected function get_offline_page_serving_script() {
-					'url'      => $image_url,
-					'revision' => $attachment->post_modified,
-				);
-			}
-		}
-
-		// Cache the theme's background image.
-		if ( current_theme_supports( 'custom-background' ) && get_background_image() ) {
-			$precache_entries[] = array(
-				// There is no attachment available, so we cannot obtain the modified date as the revision.
-				'url' => get_background_image(),
-			);
-		}
-
-		// Cache the theme's header image.
-		if ( current_theme_supports( 'custom-header' ) && get_custom_header() ) {
-			$header       = get_custom_header();
-			$attachment   = get_post( get_custom_header()->attachment_id );
-			$image_urls   = $this->get_attachment_image_urls( $attachment->ID, array( $header->width, $header->height ) );
-			$image_urls[] = get_header_image();
-
-			foreach ( array_unique( $image_urls ) as $image_url ) {
-				$precache_entries[] = array(
-					'url'      => $image_url,
-					'revision' => $attachment->post_modified,
-				);
-			}
-		}
-
-		// @todo Exclude external scripts and styles from precache?
-		wp_enqueue_scripts();
-
-		// Preload enqueued scripts.
-		wp_scripts()->all_deps( wp_scripts()->queue );
-		foreach ( wp_scripts()->to_do as $handle ) {
-			if ( ! isset( wp_scripts()->registered[ $handle ] ) ) {
-				continue;
-			}
-			$dependency = wp_scripts()->registered[ $handle ];
-
-			// Skip bundles.
-			if ( ! $dependency->src ) {
-				continue;
-			}
-
-			$src = $dependency->src;
-
-			if ( ! empty( $dependency->ver ) ) {
-				$src = add_query_arg( 'ver', $dependency->ver, $src );
-			}
-
-			/** This filter is documented in wp-includes/class.wp-scripts.php */
-			$src = apply_filters( 'script_loader_src', $src, $handle );
-
-			if ( $src ) {
-				$precache_entries[] = array(
-					'url'      => $src,
-					'revision' => $dependency->ver,
-				);
-			}
-		}
-
-		// Preload enqueued styles.
-		wp_styles()->all_deps( wp_styles()->queue );
-		foreach ( wp_styles()->to_do as $handle ) {
-			if ( ! isset( wp_styles()->registered[ $handle ] ) ) {
-				continue;
-			}
-			$dependency = wp_styles()->registered[ $handle ];
-
-			// Skip bundles.
-			if ( ! $dependency->src ) {
-				continue;
-			}
-
-			$src = $dependency->src;
-			if ( ! empty( $dependency->ver ) ) {
-				$src = add_query_arg( 'ver', $dependency->ver, $src );
-			}
-
-			/** This filter is documented in wp-includes/class.wp-styles.php */
-			$src = apply_filters( 'style_loader_src', $src, $handle );
-
-			if ( $src ) {
-				$precache_entries[] = array(
-					'url'      => $src,
-					'revision' => $dependency->ver,
-				);
-			}
-		}
-
-		return $precache_entries;
-	}
-
-	/**
-	 * Get workbox script.
+	 * @link https://github.com/GoogleChrome/workbox
 	 *
 	 * @return string Script.
 	 */
-	public function get_workbox_script() {
+	protected function get_base_script() {
 
-		$workbox_dir = 'wp-includes/js/workbox-v3.4.1/';
+		$current_scope = $this->get_current_scope();
+		$workbox_dir   = 'wp-includes/js/workbox-v3.4.1/';
```

Comment:
It actually won't be referenced anywhere else than this method. Since we define `modulePathPrefix` then the library will henceforth know how to load its own modules. Themes and plugins should not know or care where Workbox is located. So I don't think we need a constant or method in this case.